src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs

```diff
@@ -176,8 +176,15 @@ private void EnsureApplicationServices()
         {
             if (_applicationServices == null)
             {
+                _applicationServices = (IServiceProvider)_applicationServiceCollection
```

Comment:
- Doesn't this code just not call _startup.ConfigureServices(...) if an IServiceProvider is specified? 
- Any code that looks at service registrations is usually assuming too much.

Reply:
It just skips calling provider factory if instance is provided. The rest is the same as usual.

Reply:
Doesn't this code grab the IServiceProvider out of the `IServiceCollection` (which itself is an anti-pattern) and then skips calling _startup.ConfigureServices if it's an instance?